src/merge_sort.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+
+def merge_sort(a_list):
+    """Use merge sort to sort the list."""
+    if len(a_list) <= 1:
+        return a_list
+    left = []
+    right = []
+    for i in range(len(a_list)):
+        if i % 2 != 0:
+            left.append(a_list[i])
+        else:
+            right.append(a_list[i])
+    left = merge_sort(left)
+    right = merge_sort(right)
+    return _merge(left, right)
+
+
+def _merge(left, right):
+    """Takes two lists and returns single sorted list."""
+    new_list = []
+    while left and right:
+        if left[0] < right[0]:
+            new_list.append(left[0])
+            left = left[1:]
+        else:
+            new_list.append(right[0])
+            right = right[1:]
+    while left:
```

Comment:
These next 6 lines can be simplified and made much less expensive using the `list.extend` method, since all you're doing is adding the remaining elements of left, then the remaining elements of right:
```python
new_list.extend(left)
new_list.extend(right)
```